lib/optimizely/optimizely_user_context.rb

```diff
@@ -227,18 +213,24 @@ def qualified_for?(segment)
     # The segments fetched will be saved in `@qualified_segments` and can be accessed any time.
     #
     # @param options - A set of options for fetching qualified segments (optional).
-    # @return a thread handle that may be joined and will returns array of segments or nil upon error
-
-    def fetch_qualified_segments(non_blocking: false, options: [])
-      fetch_segments = lambda do |opts|
+    # @param block - An optional block to call after segments have been fetched.
+    #                If a block is provided, segments will be fetched on a separate thread.
+    #                Block will be called with a boolean indicating if the fetch succeeded.
+    # @return An array of segments or nil if fetch was unsuccessful. Method returns the thread
+    #         handle if a block is provided.
+
+    def fetch_qualified_segments(options: [], &block)
+      fetch_segments = lambda do |opts, callback|
         segments = @optimizely_client&.fetch_qualified_segments(user_id: @user_id, options: opts)
         self.qualified_segments = segments
+        callback&.call(!segments.nil?)
         segments
       end
-      if non_blocking
-        @fetch_segments_thread = Thread.new(options, &fetch_segments)
+
+      if block_given?
+        Thread.new(options, block, &fetch_segments)
       else
-        fetch_segments.call(options)
+        fetch_segments.call(options, nil)
```

Comment:
What about returning `status: boolean` for this sync call as well?  It'll be consistent with thread-callback and also we can make it clear about saving `segments` in the user-context (instead of returning the caller). Either way is acceptable, but consistent (both return `status` or `segments`) may be good. What do you think?

Reply:
I agree that the status boolean is probably more relevant to the user and they can easily retrieve the segments if desired. Also consistency is good. I've changed the sync call to return the status bool. 👍